types/supercluster/index.d.ts

```diff
@@ -4,96 +4,120 @@
      */
-    nodeSize?: number;
-    /**
+  nodeSize?: number;
+  /**
      * Whether timing info should be logged.
      */
-    log?: boolean;
+  log?: boolean;
+
+  /**
+     * initial determines the starting values for the accumulator for the reducer.
+     */
+  initial?: () => ReduceProperties;
+
+  /**
+     * map takes the properties of the geoJson properties of the point and 
+     * maps it to a MapProperties that is then passed to the reducer.
+     */
+  map?: (geoJsonProperties: any) => MapProperties;
+
+  /**
+     * reduce takes an accumulator and the mapped value of the properties and
+     * mutates the accumulator based on the value.
+     */
+  reduce?: (acc: ReduceProperties, value: MapProperties) => void;
 }
 
 export class Supercluster {
-    /**
+  /**
      * Loads an array of GeoJSON.Feature objects. Each feature's geometry must be a GeoJSON.Point. Once loaded, index is immutable.
      */
-    load(points: Points): Supercluster;
-    /**
+  load(points: Points): Supercluster;
+  /**
      * For the given bbox array ([westLng, southLat, eastLng, northLat]) and integer zoom, returns an array of clusters and points as GeoJSON.Feature objects.
      */
-    getClusters(bbox: BBox, zoom: number): Clusters;
+  getClusters(bbox: BBox, zoom: number): Clusters;
 
-    /**
+  /**
      * For a given zoom and x/y coordinates, returns a geojson-vt-compatible JSON tile object with cluster/point features.
      */
-    getTile(z: number, x: number, y: number): Tile;
+  getTile(z: number, x: number, y: number): Tile;
 
-    /**
+  /**
      * Returns the children of a cluster (on the next zoom level) given its id (cluster_id value from feature properties) and zoom the cluster was from.
      */
-    getChildren(clusterId: number, clusterZoom: number): Clusters;
+  getChildren(clusterId: number, clusterZoom: number): Clusters;
 
-    /**
+  /**
      * Returns all the points of a cluster (given its cluster_id and zoom),
      * with pagination support: limit is the number of points to return (set to Infinity for all points),
      * and offset is the amount of points to skip (for pagination).
      */
-    getLeaves(clusterId: number, clusterZoom: number, limit?: number, offset?: number): Clusters;
+  getLeaves(
+    clusterId: number,
+    clusterZoom: number,
+    limit?: number,
+    offset?: number,
+  ): Clusters;
 
-    /**
+  /**
      * Returns the zoom on which the cluster expands into several children (useful for "click to zoom" feature), given the cluster's cluster_id and zoom.
      */
-    getClusterExpansionZoom(clusterId: number, clusterZoom: number): number;
+  getClusterExpansionZoom(clusterId: number, clusterZoom: number): number;
 }
 
 /**
  * A very fast JavaScript library for geospatial point clustering for browsers and Node.
  */
-export default function supercluster(options: Options): Supercluster;
+export default function supercluster<MapProps, ReduceProps>(
```

Comment:
It would be best to define a defaults for both `MapProps` & `ReduceProps`

```ts
export default function supercluster<MapProps extends any, ReduceProps extends any>(
```

Reply:
you're definitely right have defined defaults makes it easier for the users of the library to not have to define it explicitly if they don't need it. Thanks for making the suggestion.

so turns out after running the linter it was complaining about
'''
ERROR: 97:20  no-unnecessary-generics  Type parameter MapProps is used only once. See: https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md
ERROR: 97:30  no-unnecessary-generics  Type parameter ReduceProps is used only once. See: https://github.com/Microsoft/dtslint/blob/master/docs/no-unnecessary-generics.md
'''